File: solutions/go/lasagna-master/1/lasagna_master.go
package lasagna

// TODO: define the 'PreparationTime()' function
func PreparationTime(layers []string, prepTime int) int {
	if prepTime == 0 {
		prepTime = 2
	}
	return len(layers) * prepTime
}

// TODO: define the 'Quantities()' function
func Quantities(layers []string) (int, float64) {
	noodles := 0
	sauce := 0.0
	for i := 0; i < len(layers); i++ {
		if layers[i] == "noodles" {
			noodles += 50
		}
		if layers[i] == "sauce" {
			sauce += 0.2
		}
	}
	return noodles, sauce
}

// TODO: define the 'AddSecretIngredient()' function
func AddSecretIngredient(friendsList []string, myList []string) {
	secretIngredient := friendsList[len(friendsList)-1]
	myList[len(myList)-1] = secretIngredient
}

// TODO: define the 'ScaleRecipe()' function
func ScaleRecipe(quantities []float64, portions int) []float64 {
	var res []float64
	for i := 0; i < len(quantities); i++ {
		res = append(res, (float64(portions)*quantities[i])/2.0)
	}
	return res
}

// Your first steps could be to read through the tasks, and create
// these functions with their correct parameter lists and return types.
// The function body only needs to contain `panic("")`.
//
// This will make the tests compile, but they will fail.
// You can then implement the function logic one by one and see
// an increasing number of tests passing as you implement more
// functionality.

File: solutions/cpp/lasagna-master/1/lasagna_master.cpp
#include "lasagna_master.h"
#include <string>
#include <vector>
#include <cstddef>

namespace lasagna_master {

// TODO: add your solution here
    int preparationTime(const std::vector<std::string>& layers, const int avg_preparation_time) {
        return layers.size() * avg_preparation_time;
    }

    amount quantities(const std::vector<std::string>& layers) {
        amount result {0, 0.0};

        for (std::string layer : layers) {
            if (layer == "noodles") {
                result.noodles += 50;
            } else if (layer == "sauce") {
                result.sauce += 0.2;
            }
        }

        return result;
    }

    void addSecretIngredient(std::vector<std::string>& yourRecipe, const std::vector<std::string>& friendsRecipe) {
        yourRecipe[yourRecipe.size() - 1] = friendsRecipe[friendsRecipe.size() - 1];
    }

    std::vector<double> scaleRecipe(const std::vector<double>& quantities, int portions) {
        double multiplier = portions / 2.0;
        std::vector<double> res {};

        for (std::size_t i = 0; i < quantities.size(); ++i) {
            res.emplace_back(quantities[i] * multiplier);
        }

        return res;
    }

    void addSecretIngredient(std::vector<std::string>& yourRecipe, std::string secretIngredient) {
        yourRecipe[yourRecipe.size() - 1] = secretIngredient;
    }
}  // namespace lasagna_master
